src/Nethermind/Nethermind.Synchronization/SnapSync/ProgressTracker.cs

```diff
@@ -23,6 +26,11 @@ public class ProgressTracker
         private const int CODES_BATCH_SIZE = 1_000;
         private readonly byte[] ACC_PROGRESS_KEY = Encoding.ASCII.GetBytes("AccountProgressKey");
 
+        // This does not need to be a lot as it spawn other requests. In fact 8 is probably too much. It is severely
+        // bottlenecked by _syncCommit lock in SnapProviderHelper, which in turns is limited by the IO.
+        // In any case, all partition will be touched when calculating progress, so we can't really put like 1024 for this.
+        private readonly int _accountRangePartitionCount;
```

Comment:
what would happen if we could remove `_syncCommit` lock somehow?

Reply:
If I recall correctly, just removing the lock will eventually causes error. Did not remember what. As mentioned in the description, I tried to just create a new trie store on each request then remove the lock. Seems to work just fine. I just dont include it here. Using object pool probably would also work. In theory, some of the stitch might get missed whuch causes some of the inner node to not get saved, but its probably not gonna be valid, and healing should fix it.